roles/update/tasks/update_run.yml

```diff
@@ -0,0 +1,196 @@
+---
+- name: Wait for MinorUpdateAvailable condition available
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    api_version: core.openstack.org/v1beta1
+    kind: OpenStackVersion
+    namespace: "{{ cifmw_update_namespace }}"
+  register: _openstackversion_info
+  retries: 20
+  delay: 15
+  until:
+    - _openstackversion_info.resources[0].status.conditions is defined
+    - >-
+      _openstackversion_info.resources[0].status.conditions |
+      selectattr("type", "equalto", "MinorUpdateAvailable") |
+      selectattr("status", "equalto", "True") | length > 0
+
+# start update by changing targetVersion to new availableVersion
+- name: Change targetVersion to availableVersion
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.shell: >-
+    oc patch openstackversion
+    $(oc get openstackversion
+    -o=jsonpath='{.items[].metadata.name}')
+    --type merge --patch '{"spec":{"targetVersion":_cifmw_update_openstack_version}}'
+
+- name: Wait for ovn update completion on control plane
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    api_version: core.openstack.org/v1beta1
+    kind: OpenStackVersion
+    namespace: "{{ cifmw_update_namespace }}"
+  register: _openstackversion_info
+  retries: 20
+  delay: 15
+  until:
+    - _openstackversion_info.resources[0].status.conditions is defined
+    - >-
+      _openstackversion_info.resources[0].status.conditions |
+      selectattr("type", "equalto", "MinorUpdateOVNControlplane") |
+      selectattr("status", "equalto", "True") | length > 0
+
+- name: Fetch NodeSets for the OVN update run
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path | default(ansible_env.PATH) }}"
+  ansible.builtin.shell: >-
+    set -o pipefail;
+    nodes_with_ovn=()
+    # Get the names of all OpenStackDataPlaneNodeSet resources
+    openstackdataplanenodesets=$(oc get openstackdataplanenodeset -o custom-columns=NAME:.metadata.name,SERVICES:.spec.services --no-headers)
+    # Loop through each OpenStackDataPlaneNodeSet
+    while read -r node_name services; do
```

Comment:
I understand - it would work, but... why? Why not doing that in 2 tasks and use ansible loop for that?